### PR TITLE
Upgrade Spotless and Ktlint and remove code convention violations

### PR DESCRIPTION
**Pull request by lihenggui**
**What I have done and why**
Upgrade Spotless from 6.22.0 to [6.23.3](https://github.com/diffplug/spotless/blob/main/plugin-gradle/CHANGES.md#6233---2023-12-04)
Upgrade ktlint from 0.48.1 to [1.0.1](https://github.com/pinterest/ktlint/blob/master/CHANGELOG.md#101---2023-10-13)
Fixed all spotless errors in the project
[Remove violations: discouraged-comment-location](https://github.com/android/nowinandroid/commit/f8f932705a71d47f4925f5cad9bbbcacdb3cc0c1)
[Remove violations: property-naming](https://github.com/android/nowinandroid/commit/32e1ec0447294ada76c83bc1f54a88f3487b0e04)

**Do tests pass?**
- [ Y ] Run local tests on `DemoDebug` variant: `./gradlew testDemoDebug`
- [ Y ] Check formatting: `./gradlew --init-script gradle/init.gradle.kts spotlessApply`

### CODE REVIEW

**Review by dturner — COMMENTED** [review]
Thanks for doing this.

File: benchmarks/src/main/kotlin/com/google/samples/apps/nowinandroid/startup/StartupBenchmark.kt [hunk 1]
```diff
@@ -60,7 +60,8 @@ class StartupBenchmark {
         packageName = PACKAGE_NAME,
         metrics = listOf(StartupTimingMetric()),
         compilationMode = compilationMode,
-        iterations = 20, // More iterations result in higher statistical significance.
+        // More iterations result in higher statistical significance.
```
[the remark this answers is not in the document]
Comment (lihenggui) [reply benchmarks/src/main/kotlin/com/google/samples/apps/nowinandroid/startup/StartupBenchmark.kt:63]:
```
rule: [standard:discouraged-comment-location](https://pinterest.github.io/ktlint/1.1.0/rules/standard/#discouraged-comment-location)
A comment in a 'value_argument_list' is only allowed when placed on a separate line
```
Reason: https://pinterest.github.io/ktlint/latest/faq/#why-does-ktlint-discourage-certain-comment-locations

> In other cases, a comment location is more widely used but semantically still incorrect. For example, in sample below the EOL comment is placed after the comma, but it obviously is related to the part before the comma:
> 
> 
>  Unclear comment
>  Clear comment
> fun fooBar(
>     foo: Foo, // foo-comment
>     bar: Bar, // bar-comment
> ) {}
> 
> By forbidding certain comment locations, the logic in the rules becomes a bit easier.


We can disable this rule in the config file if we want to keep this style of comment.

File: core/network/src/test/kotlin/com/google/samples/apps/nowinandroid/core/network/fake/FakeNiaNetworkDataSourceTest.kt [hunk 2]
```diff
@@ -45,46 +45,46 @@ class FakeNiaNetworkDataSourceTest {
...
-                longDescription = "The latest events and announcements from the world of Android development.",
-                url = "",
-                imageUrl = "https://firebasestorage.googleapis.com/v0/b/now-in-android.appspot.com/o/img%2Fic_topic_Headlines.svg?alt=media&token=506faab0-617a-4668-9e63-4a2fb996603f",
-            ),
-            /* ktlint-enable max-line-length */
-            subject.getTopics().first(),
-        )
-    }
+    fun testDeserializationOfTopics() =
+        @Suppress("ktlint:standard:max-line-length")
```
Comment (SimonMarquis) [inline core/network/src/test/kotlin/com/google/samples/apps/nowinandroid/core/network/fake/FakeNiaNetworkDataSourceTest.kt:49]: nit: Can this be moved on the function scope itself?

```
@Suppress("ktlint:standard:max-line-length")
@Test
fun testDeserializationOfTopics() = runTest(testDispatcher) {
```

File: core/common/src/test/kotlin/com/google/samples/apps/nowinandroid/core/result/ResultKtTest.kt [hunk 3]
```diff
@@ -25,7 +25,7 @@ import kotlin.test.assertEquals
 class ResultKtTest {
 
     @Test
-    fun Result_catches_errors() = runTest {
+    fun result_catches_errors() = runTest {
```
Comment (SimonMarquis) [inline core/common/src/test/kotlin/com/google/samples/apps/nowinandroid/core/result/ResultKtTest.kt:28]: nit: ktlint does not understand that we are referencing actual class names here. I think we should keep them. (same for the next files)

File: app/src/main/kotlin/com/google/samples/apps/nowinandroid/navigation/NiaNavHost.kt [hunk 4]
```diff
@@ -20,7 +20,7 @@ import androidx.compose.runtime.Composable
 import androidx.compose.ui.Modifier
 import androidx.navigation.compose.NavHost
 import com.google.samples.apps.nowinandroid.feature.bookmarks.navigation.bookmarksScreen
-import com.google.samples.apps.nowinandroid.feature.foryou.navigation.forYouNavigationRoute
+import com.google.samples.apps.nowinandroid.feature.foryou.navigation.FOR_YOU_NAVIGATION_ROUTE
```
[the remark this answers is not in the document]
Comment (lihenggui) [reply app/src/main/kotlin/com/google/samples/apps/nowinandroid/navigation/NiaNavHost.kt]: Changed in https://github.com/android/nowinandroid/pull/1094/files/32e1ec0447294ada76c83bc1f54a88f3487b0e04..2a2d8ed88df563e22e82b0b53efa351850f3c629

File: core/network/src/test/kotlin/com/google/samples/apps/nowinandroid/core/network/fake/FakeNiaNetworkDataSourceTest.kt (hunk 2 above)
Comment (lihenggui) [reply core/network/src/test/kotlin/com/google/samples/apps/nowinandroid/core/network/fake/FakeNiaNetworkDataSourceTest.kt]: Updated: d671126f9eb5532206db2c7260e3cc06a978dd77

File: core/common/src/test/kotlin/com/google/samples/apps/nowinandroid/core/result/ResultKtTest.kt (hunk 3 above)
Comment (lihenggui) [reply core/common/src/test/kotlin/com/google/samples/apps/nowinandroid/core/result/ResultKtTest.kt:28]: Shall we follow the `given_when_then` or `when_then` format?

Comment (SimonMarquis) [reply core/common/src/test/kotlin/com/google/samples/apps/nowinandroid/core/result/ResultKtTest.kt:28]: Or use `ktlint_function_naming_ignore_when_annotated_with` config for now and rename later if needed.

Comment (SimonMarquis) [reply core/common/src/test/kotlin/com/google/samples/apps/nowinandroid/core/result/ResultKtTest.kt]: There are other instances of this in other test files, that's why I recommended to use the .editorconfig parameter :)

Comment (lihenggui) [reply core/common/src/test/kotlin/com/google/samples/apps/nowinandroid/core/result/ResultKtTest.kt]: Please review: 56fbe13b5abe497df8b32a262001a922f3f9921c

File: benchmarks/src/main/kotlin/com/google/samples/apps/nowinandroid/startup/StartupBenchmark.kt (hunk 1 above)
[the remark this answers is not in the document]
Comment (dturner) [reply benchmarks/src/main/kotlin/com/google/samples/apps/nowinandroid/startup/StartupBenchmark.kt:63]: Thanks for the clarification. That makes sense. Let's keep the rule.

**Review by dturner — APPROVED** [review]

**Pull request merged**